packages/integrations/cloudflare/src/index.ts

```diff
@@ -105,17 +206,49 @@ export default function createIntegration(args?: Options): AstroIntegration {
 				_buildConfig = config.build;
 
 				if (config.output === 'static') {
-					throw new Error(`
-  [@astrojs/cloudflare] \`output: "server"\` or \`output: "hybrid"\` is required to use this adapter. Otherwise, this adapter is not necessary to deploy a static site to Cloudflare.
-
-`);
+					throw new Error(STATIC_ERROR);
 				}
 
 				if (config.base === SERVER_BUILD_FOLDER) {
-					throw new Error(`
-  [@astrojs/cloudflare] \`base: "${SERVER_BUILD_FOLDER}"\` is not allowed. Please change your \`base\` config to something else.`);
+					throw new Error(FOLDER_ERROR);
 				}
 			},
+			'astro:server:setup': ({ server }) => {
+				server.middlewares.use(async function middleware(req, res, next) {
+					if (runtimeMode === 'off') {
+						next();
+					} else {
+						const vars = await CFVars();
+						const cf = await getCFObject(runtimeMode);
+
+						const clientLocalsSymbol = Symbol.for('astro.locals');
+						Reflect.set(req, clientLocalsSymbol, {
+							runtime: {
+								env: {
+									// default binding for static assets will be dynamic once we support mocking of bindings
+									ASSETS: {},
+									// this is just a VAR for CF to change build behavior, on dev it should be 0
+									CF_PAGES: '0',
+									// will be fetched from git dynamically once we support mocking of bindings
+									CF_PAGES_BRANCH: 'TBA',
+									// will be fetched from git dynamically once we support mocking of bindings
+									CF_PAGES_COMMIT_SHA: 'TBA',
+									CF_PAGES_URL: `http://${req.headers.host}`,
+									...vars,
+								},
+								cf: cf,
+								// will be supported once we support mocking of bindings
+								// waitUntil: (promise: Promise<any>) => {
+								//   context.waitUntil(promise);
+								// },
```

Comment:
I think it would be appropriate for this to be a no-op.

Reply:
Sorry, I'm at a loss, what exactly do you mean by no-op?

Reply:
.waitUntil() only makes sense for serverless environments, where the instance needs to be told to stay awake.

Astro's dev server is going to stay up anyway, so this function can be a non-operation, a function that does nothing.